File: src/main/java/com/morak/performancetracker/aop/persistence/ConnectionPointcut.java
package com.morak.performancetracker.aop.persistence;

import java.lang.reflect.Method;
import org.springframework.aop.support.StaticMethodMatcherPointcut;

public class ConnectionPointcut extends StaticMethodMatcherPointcut {

    private static final String METHOD_SUFFIX = "prepareStatement";

    @Override
    public boolean matches(Method method, Class<?> targetClass) {
        return method.getName().endsWith(METHOD_SUFFIX);
    }
}

File: src/main/kotlin/com/morak/performancetracker/aop/persistence/ConnectionPointcut.kt
package com.morak.performancetracker.aop.persistence

import org.springframework.aop.support.StaticMethodMatcherPointcut
import java.lang.reflect.Method

class ConnectionPointcut : StaticMethodMatcherPointcut() {
    override fun matches(method: Method, targetClass: Class<*>?): Boolean {
        return method.name.endsWith(METHOD_SUFFIX)
    }

    companion object {
        private const val METHOD_SUFFIX = "prepareStatement"
    }
}
